Library/Homebrew/cmd/uninstall.rb

```diff
@@ -87,36 +87,64 @@ def check_for_dependents(kegs)
     return false unless result = Keg.find_some_installed_dependents(kegs)
 
     if ARGV.homebrew_developer?
-      dependents_output_for_developers(*result)
+      DeveloperDependentsMessage.new(*result).output
     else
-      dependents_output_for_nondevelopers(*result)
+      NondeveloperDependentsMessage.new(*result).output
     end
 
     true
   end
 
-  def dependents_output_for_developers(requireds, dependents)
-    msg = requireds.join(", ")
-    msg << (requireds.count == 1 ? " is" : " are")
-    msg << " required by #{dependents.join(", ")}, which "
-    msg << (dependents.count == 1 ? "is" : "are")
-    msg << " currently installed."
-    msg << "\nYou can silence this warning with "
-    msg << "`brew uninstall --ignore-dependencies "
-    msg << "#{requireds.map(&:name).join(" ")}`."
-    opoo msg
+  class DependentsMessage
```

Comment:
Put all the shared stuff between the two messages into a class so it can be re-used between both of them. It's a bit longer, but I think much cleaner now.